sql/src/main/java/io/crate/analyze/Fields.java

```diff
@@ -57,6 +61,26 @@ public Field get(ColumnIdent key) {
         return fieldList.iterator().next();
     }
 
+    @Nullable
+    public Field getWithSubscriptFallback(ColumnIdent column,
+                                          AnalyzedRelation scope,
+                                          AnalyzedRelation childRelation,
+                                          boolean pointToChild) {
```

Comment:
This `pointToChild` argument could probably use some documentation that explains a bit why one would set it to true vs to false.